dashboard/test/ui/step_definitions/steps.rb

```diff
@@ -10,9 +10,8 @@ def replace_hostname(url)
     url = url.gsub(/\/\/code.org\//, "//" + ENV['PEGASUS_TEST_DOMAIN'] + "/")
   end
   # Convert http to https, and x.y.code.org to x-y.code.org
-  url.
-    gsub(/^http:\/\//,'https://').
-    gsub(/(\w+)\.(\w+)\.code\.org/,'\1-\2.code.org')
+  url.gsub(/^http:\/\//,'https://') unless url.starts_with? 'http://localhost'
```

Comment:
i think this should become `gsub!` or `url = url.gsub`, if it's broken into a separate statement like this
`gsub!` modifies string in-place, `gsub` returns a copy of the modified string